From 575ea9f2412905a323cd0c3c380f003bb9e61e67 Mon Sep 17 00:00:00 2001
From: Jim Newsome <jnewsome@torproject.org>
Date: Sun, 3 May 2020 10:09:10 -0500
Subject: [PATCH 2/2] Hack to avoid crash when using as a preloaded library

---
 malloc/arena.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/malloc/arena.c b/malloc/arena.c
index cecdb7f4c4..8c3fc8c867 100644
--- a/malloc/arena.c
+++ b/malloc/arena.c
@@ -296,12 +296,14 @@ ptmalloc_init (void)
 #ifdef SHARED
   /* In case this libc copy is in a non-default namespace, never use brk.
      Likewise if dlopened from statically linked program.  */
+  /*
   Dl_info di;
   struct link_map *l;
 
   if (_dl_open_hook != NULL
       || (_dl_addr (ptmalloc_init, &di, &l, NULL) != 0
           && l->l_ns != LM_ID_BASE))
+          */
     __morecore = __failing_morecore;
 #endif
 
-- 
2.17.1

